lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -0,0 +1,233 @@
+import pytest
+import requests
+from http import HTTPStatus
+from typing import Any, AnyStr, Dict
+
+
+class Commons:
+    """
+    Unit testing for all the elasticsearch resources class.
+
+    In a web service context, we access class functions mostly via the
+    Flask test client rather than trying to directly invoke the class
+    constructor and `post` service.
+    """
+
+    def _setup(
+        self,
+        pbench_endpoint: AnyStr = None,
+        elastic_endpoint: AnyStr = None,
+        payload: [AnyStr, Any] = None,
+        bad_date_payload: [AnyStr, Any] = None,
+        error_payload: Dict[AnyStr, Any] = None,
+        empty_response_payload: Dict[AnyStr, Any] = None,
+    ):
+        self.pbench_endpoint = pbench_endpoint
+        self.elastic_endpoint = elastic_endpoint
+        self.payload = payload
+        self.bad_date_payload = bad_date_payload
+        self.error_payload = error_payload
+        self.empty_response_payload = empty_response_payload
+
+    def build_index(self, server_config, dates):
+        """
+        Build the index list for query
+
+        Args:
+            dates (iterable): list of date strings
+        """
+        idx = server_config.get("Indexing", "index_prefix") + ".v6.run-data."
+        index = "/"
+        for d in dates:
+            index += f"{idx}{d},"
+        return index
+
+    def test_non_accessible_user_data(self, client, server_config, pbench_token):
+        """
+        Test behavior when Authorization header does not have access to other user's data
+        """
+        # The pbench_token fixture logs in as user "drb"
+        # Trying to access the data belong to the user "pp"
+        if not self.payload.get("user", None):
+            pytest.skip("skipping non accessible user data test")
```

Comment:
> This at least deserves an explanation in the docstring; you're skipping out if the test doesn't have `"user"` or if it's given as `None`, but then overwriting whatever value was given with `"pp"` below. So we _have_ to supply it, or the call does nothing, but if we _do_ supply it, the value we specified isn't used ... ???

Yes because as the name of the test suggest its testing the non accessible user data, the pbench_token that we use for  this test belongs to the "drb" user who will try to access the user with username "pp" (comment on line 49-50) and that's why I deliberately change the payload user value to "pp"

> I'm assuming this is to deactivate the test case for publish, which doesn't have a `user` parameter and requires `Dataset` setup to test authorization failures; and assumes that all the other subclasses will supply `user` in their default payload parameter to the super-constructor. It'd be cleaner to rely on the subclass schema, which you appear to be accessing in other contexts.

yes that good idea I will try that